From 35f84e1791eaae8c8c0f4798a5cba3061277a672 Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Wed, 17 Apr 2024 17:03:56 +0100
Subject: [PATCH] DTS: overlays: fix Pi 5 midi-over-UART

The UART source clock is 50MHz not 100MHz, so fix the fake-clock used to
munge the baudrate.

See https://forums.raspberrypi.com/viewtopic.php?p=2212755

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/midi-uart0-pi5-overlay.dts | 6 +++---
 arch/arm/boot/dts/overlays/midi-uart1-pi5-overlay.dts | 6 +++---
 arch/arm/boot/dts/overlays/midi-uart2-pi5-overlay.dts | 6 +++---
 arch/arm/boot/dts/overlays/midi-uart3-pi5-overlay.dts | 6 +++---
 arch/arm/boot/dts/overlays/midi-uart4-pi5-overlay.dts | 6 +++---
 5 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/midi-uart0-pi5-overlay.dts b/arch/arm/boot/dts/overlays/midi-uart0-pi5-overlay.dts
index 6cd1f3ed2d8d..837d1b014e28 100644
--- a/arch/arm/boot/dts/overlays/midi-uart0-pi5-overlay.dts
+++ b/arch/arm/boot/dts/overlays/midi-uart0-pi5-overlay.dts
@@ -5,10 +5,10 @@
 
 /*
  * Fake a higher clock rate to get a larger divisor, and thereby a lower
- * baudrate. The real clock is 100MHz, which we scale so that requesting
+ * baudrate. The real clock is 50MHz, which we scale so that requesting
  * 38.4kHz results in an actual 31.25kHz.
  *
- *   100000000*38400/31250 = 122880000
+ *   50000000*38400/31250 = 61440000
  */
 
 /{
@@ -21,7 +21,7 @@ midi_clk: midi_clk0 {
 				compatible = "fixed-clock";
 				#clock-cells = <0>;
 				clock-output-names = "uart0_pclk";
-				clock-frequency = <122880000>;
+				clock-frequency = <61440000>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/midi-uart1-pi5-overlay.dts b/arch/arm/boot/dts/overlays/midi-uart1-pi5-overlay.dts
index 18f526865eed..e803876622a9 100644
--- a/arch/arm/boot/dts/overlays/midi-uart1-pi5-overlay.dts
+++ b/arch/arm/boot/dts/overlays/midi-uart1-pi5-overlay.dts
@@ -5,10 +5,10 @@
 
 /*
  * Fake a higher clock rate to get a larger divisor, and thereby a lower
- * baudrate. The real clock is 100MHz, which we scale so that requesting
+ * baudrate. The real clock is 50MHz, which we scale so that requesting
  * 38.4kHz results in an actual 31.25kHz.
  *
- *   100000000*38400/31250 = 122880000
+ *   50000000*38400/31250 = 61440000
  */
 
 /{
@@ -21,7 +21,7 @@ midi_clk: midi_clk1 {
 				compatible = "fixed-clock";
 				#clock-cells = <0>;
 				clock-output-names = "uart1_pclk";
-				clock-frequency = <122880000>;
+				clock-frequency = <61440000>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/midi-uart2-pi5-overlay.dts b/arch/arm/boot/dts/overlays/midi-uart2-pi5-overlay.dts
index 5e1e0c6fd7a9..4f07e7de2df3 100644
--- a/arch/arm/boot/dts/overlays/midi-uart2-pi5-overlay.dts
+++ b/arch/arm/boot/dts/overlays/midi-uart2-pi5-overlay.dts
@@ -5,10 +5,10 @@
 
 /*
  * Fake a higher clock rate to get a larger divisor, and thereby a lower
- * baudrate. The real clock is 100MHz, which we scale so that requesting
+ * baudrate. The real clock is 50MHz, which we scale so that requesting
  * 38.4kHz results in an actual 31.25kHz.
  *
- *   100000000*38400/31250 = 122880000
+ *   50000000*38400/31250 = 61440000
  */
 
 /{
@@ -21,7 +21,7 @@ midi_clk: midi_clk2 {
 				compatible = "fixed-clock";
 				#clock-cells = <0>;
 				clock-output-names = "uart2_pclk";
-				clock-frequency = <122880000>;
+				clock-frequency = <61440000>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/midi-uart3-pi5-overlay.dts b/arch/arm/boot/dts/overlays/midi-uart3-pi5-overlay.dts
index 705a2793d00c..478220d41edc 100644
--- a/arch/arm/boot/dts/overlays/midi-uart3-pi5-overlay.dts
+++ b/arch/arm/boot/dts/overlays/midi-uart3-pi5-overlay.dts
@@ -5,10 +5,10 @@
 
 /*
  * Fake a higher clock rate to get a larger divisor, and thereby a lower
- * baudrate. The real clock is 100MHz, which we scale so that requesting
+ * baudrate. The real clock is 50MHz, which we scale so that requesting
  * 38.4kHz results in an actual 31.25kHz.
  *
- *   100000000*38400/31250 = 122880000
+ *   50000000*38400/31250 = 61440000
  */
 
 /{
@@ -21,7 +21,7 @@ midi_clk: midi_clk3 {
 				compatible = "fixed-clock";
 				#clock-cells = <0>;
 				clock-output-names = "uart3_pclk";
-				clock-frequency = <122880000>;
+				clock-frequency = <61440000>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/midi-uart4-pi5-overlay.dts b/arch/arm/boot/dts/overlays/midi-uart4-pi5-overlay.dts
index 0d2f823ed7dd..827bd5e951ba 100644
--- a/arch/arm/boot/dts/overlays/midi-uart4-pi5-overlay.dts
+++ b/arch/arm/boot/dts/overlays/midi-uart4-pi5-overlay.dts
@@ -5,10 +5,10 @@
 
 /*
  * Fake a higher clock rate to get a larger divisor, and thereby a lower
- * baudrate. The real clock is 100MHz, which we scale so that requesting
+ * baudrate. The real clock is 50MHz, which we scale so that requesting
  * 38.4kHz results in an actual 31.25kHz.
  *
- *   100000000*38400/31250 = 122880000
+ *   50000000*38400/31250 = 61440000
  */
 
 /{
@@ -21,7 +21,7 @@ midi_clk: midi_clk4 {
 				compatible = "fixed-clock";
 				#clock-cells = <0>;
 				clock-output-names = "uart4_pclk";
-				clock-frequency = <122880000>;
+				clock-frequency = <61440000>;
 			};
 		};
 	};
-- 
2.34.1

